tests/pjit_test.py

```diff
@@ -201,8 +201,68 @@ def f(x):
     # Annotation from pjit
     self.assertIn("sharding={replicated}", hlo.as_hlo_text())
 
-
   # TODO(skye): add more unit tests once API is more finalized
 
+
+class PJitErrorTest(jtu.JaxTestCase):
+  @ignore_pjit_warning()
+  @with_mesh([('x', 2)])
+  def testNonDivisibleArgs(self):
+    x = jnp.ones((3,))
+    with self.assertRaisesRegex(ValueError,
+                                r"One of pjit arguments.*PartitionSpec\('x', None\).*"
+                                r"implies that the size of its dimension 0 should be "
+                                r"divisible by 2, but it is equal to 3"):
```

Comment:
nit: I'm a big fan of including complete error messages in tests, because it makes it easier to verify that the error message is clear and properly formatted (both for the test writer and future readers). Up to you if you wanna do that here and below.

Reply:
I usually do this just to ensure that we put in the key information I'd like to see while debugging in the error message. After all, you can check the formatting in the original string for the message, so I'll leave it as is. But feel free to modify in a follow-up if you feel like you want to have full messages here.